Failure cluster [d7f6dc54...] `Failed to get successful response from /configz: context deadline exceeded`

### Failure cluster [d7f6dc54a335e039ff4b](https://go.k8s.io/triage#d7f6dc54a335e039ff4b)

<img width="3344" height="1072" alt="Image" src="https://github.com/user-attachments/assets/abc74568-47c6-4c7c-97a9-ab782f238481" />

##### Error text:
```
[FAILED] Failed to get successful response from /configz: context deadline exceeded
In [It] at: k8s.io/kubernetes/test/e2e_node/kubeletconfig/kubeletconfig.go:186 @ 05/30/26 05:26:56.65

```
#### Recent failures:
[5/30/2026, 10:22:10 AM ci-node-crio-resource-managers](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-node-crio-resource-managers/2060728360425754624)
[5/30/2026, 9:50:11 AM ci-kubernetes-node-kubelet-containerd-resource-managers](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-node-kubelet-containerd-resource-managers/2060720307273994240)
[5/30/2026, 8:29:11 AM ci-containerd-node-e2e-serial-ec2](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-containerd-node-e2e-serial-ec2/2060699671034073088)
[5/30/2026, 7:58:11 AM ci-kubernetes-node-swap-ubuntu-serial](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-node-swap-ubuntu-serial/2060692121802772480)
[5/30/2026, 7:03:55 AM ci-node-crio-swap-serial](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-node-crio-swap-serial/2060677777220177920)


/kind failing-test
<!-- If this is a flake, please add: /kind flake -->

/sig node

## Root cause: #139351 (locking `DisableCPUQuotaWithExclusiveCPUs` to default) — kubelet crash-loop, not a flake

This is a hard regression, not a flake. The `/configz` timeout is a symptom: the kubelet **panics on startup** and crash-loops, so `GetCurrentKubeletConfig` (which polls `https://127.0.0.1:10250/configz` for 5m in the setup of nearly every Serial/resource-manager test) times out with `context deadline exceeded`.

### Regression boundary (first bad build)
Both resource-manager lanes flipped green→red at the same SHAs:

| lane | last green | first red |
|---|---|---|
| `…containerd-resource-managers` | [`a3242d59`](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-node-kubelet-containerd-resource-managers/2060478461952135168) (2026-05-29 21:49Z) | [`bb7dec4604f`](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-node-kubelet-containerd-resource-managers/2060538860126867456) (2026-05-30 01:49Z) |
| `ci-node-crio-resource-managers` | [`a3242d59`](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-node-crio-resource-managers/2060486514827071488) (2026-05-29 22:21Z) | [`bb7dec4604f`](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-node-crio-resource-managers/2060546913320570880) (2026-05-30 02:21Z) |

Commit window `a3242d59..bb7dec4604f` contained 3 PRs; **#139351** is the cause.

### Mechanism
`kubelet.log` from the first red build ends in:

```
panic: failed to merge global and in-flight KubeletConfiguration while setting defaults,
error: cannot set feature gate DisableCPUQuotaWithExclusiveCPUs to false, feature is locked to true

goroutine 1 [running]:
…SetDefaults_KubeletConfiguration  pkg/kubelet/apis/config/v1beta1/defaults.go:72
…DecodeKubeletConfiguration        pkg/kubelet/kubeletconfig/util/codec/codec.go:75
…loadConfigFile                    cmd/kubelet/app/server.go:464
```

1. #139351 added `{Version: "1.37", Default: true, LockToDefault: true}` to `DisableCPUQuotaWithExclusiveCPUs`, so the gate can no longer be set to `false`.
2. `SetDefaults_KubeletConfiguration` (`defaults.go:72`) does `localFeatureGate.SetFromMap(obj.FeatureGates)` and **`panic()`s** on the locked-gate conflict when the kubelet decodes its config file at startup.
3. The CPU Manager e2e test writes the gate unconditionally — `configureCPUManagerInKubelet` (`test/e2e_node/cpu_manager_test.go:4034`) — and the block `When("checking the CFS quota management can be disabled")` sets `disableCPUQuotaWithExclusiveCPUs: false` in its `BeforeEach` (`cpu_manager_test.go:1747`). Restarting the kubelet with that config → panic → crash-loop. The `true` cases are fine (that's the locked default); only the `false` case panics.

Because these are **Serial** suites sharing one kubelet, the wedged node cascades the failure to subsequent CPU/Memory/Topology Manager tests in the same run.

### Confirmed across 4 jobs — runtime/arch/provider independent (rules out infra)
The identical panic appears in:

| job | runtime | arch | provider | first red |
|---|---|---|---|---|
| `ci-kubernetes-node-kubelet-containerd-resource-managers` | containerd | x86 | GCE | [2060538860126867456](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-node-kubelet-containerd-resource-managers/2060538860126867456) |
| `ci-node-crio-resource-managers` | **CRI-O** | x86 | GCE | [2060546913320570880](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-node-crio-resource-managers/2060546913320570880) |
| `ci-containerd-node-e2e-serial-ec2` | containerd | x86 | **AWS EC2** | [2060518224151187456](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-containerd-node-e2e-serial-ec2/2060518224151187456) |
| `ci-containerd-node-arm64-e2e-serial-ec2` | containerd | **ARM64** | AWS EC2 | [2060536595643109376](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-containerd-node-arm64-e2e-serial-ec2/2060536595643109376) |

### Why presubmit didn't catch it
These node-e2e Serial/resource-manager lanes are periodic post-submit only. #139351's presubmits passed — it updated the unit test (`kuberuntime_container_linux_test.go`) but not the `e2e_node` CPU Manager test that sets the gate to `false`.

### Suggested fix
Same class as #137996 (which did exactly this for `InPlacePodLevelResourcesVerticalScaling`); background in #127477. Either:
- update `test/e2e_node/cpu_manager_test.go` to stop setting `DisableCPUQuotaWithExclusiveCPUs: false` (drop/guard the "CFS quota can be disabled" scenario now that the gate is locked to `true`), or
- revert #139351 until the e2e test is updated.

Separately worth considering: `defaults.go:72` turning a bad gate into a `panic()` crash-loops the kubelet instead of failing with a clean validation error.

Earlier same-signature flake issue (pre-dates this regression): #137995.

/cc @ffromani

--- a/test/e2e_node/cpu_manager_test.go
+++ b/test/e2e_node/cpu_manager_test.go
@@ -1737,6 +1737,7 @@ var _ = SIGDescribe("CPU Manager", ginkgo.Ordered, ginkgo.ContinueOnFailure, fra
 		// don't duplicate the all the tests
 
 		ginkgo.BeforeEach(func(ctx context.Context) {
+			e2eskipper.Skipf("DisableCPUQuotaWithExclusiveCPUs is locked to its default (true); enforcing CFS quota for exclusive CPUs is no longer configurable")
 			requireCGroupV2()
 			// WARNING: this assumes 2-way SMT systems - we don't know how to access other SMT levels.
 			//          this means on more-than-2-way SMT systems this test will prove nothing
@@ -4031,7 +4032,11 @@ func configureCPUManagerInKubelet(oldCfg *kubeletconfig.KubeletConfiguration, ku
 
 	newCfg.FeatureGates["CPUManagerPolicyBetaOptions"] = kubeletArguments.enableCPUManagerOptions
 	newCfg.FeatureGates["CPUManagerPolicyAlphaOptions"] = kubeletArguments.enableCPUManagerOptions
-	newCfg.FeatureGates["DisableCPUQuotaWithExclusiveCPUs"] = kubeletArguments.disableCPUQuotaWithExclusiveCPUs
+	// DisableCPUQuotaWithExclusiveCPUs is locked to its default (true); the
+	// kubelet rejects setting it to false, so only set it when true.
+	if kubeletArguments.disableCPUQuotaWithExclusiveCPUs {
+		newCfg.FeatureGates["DisableCPUQuotaWithExclusiveCPUs"] = true
+	}
 	newCfg.FeatureGates["PodLevelResources"] = kubeletArguments.enablePodLevelResources
 	// InPlacePodLevelResourcesVerticalScaling is only supported when PodLevelResources is enabled
 	if !kubeletArguments.enablePodLevelResources {

